File: src/main/java/com/easipass/epUtil/core/entity/dto/CusResultDTO.java
package com.easipass.epUtil.core.entity.dto;

import com.zj0724.springbootUtil.annotation.NotNull;

/**
 * 回执DTO
 *
 * @author ZJ
 * */
public class CusResultDTO {

    /**
     * 状态
     * */
    @NotNull
    private String channel;

    /**
     * 备注
     * */
    @NotNull
    private String note;

    public CusResultDTO() {}

    public CusResultDTO(String channel, String note) {
        this.channel = channel;
        this.note = note;
    }

    public String getChannel() {
        return channel;
    }

    public void setChannel(String channel) {
        this.channel = channel;
    }

    public String getNote() {
        return note;
    }

    public void setNote(String note) {
        this.note = note;
    }

    @Override
    public String toString() {
        return "CusResultDTO{" +
                "channel='" + channel + '\'' +
                ", note='" + note + '\'' +
                '}';
    }

}

File: src/main/ui/entity/DTO/CusResultDTO.js
(function (DTO) {

    DTO.CusResultDTO = class CusResultDTO {

        channel = "";

        note = "";

        // 校验
        check(callback) {
            if (this.channel === "") {
                callback("channel不能为空");
                return false;
            }

            if (this.note === "") {
                callback("note不能为空");
                return false;
            }

            return true;
        }

    }

})(window.epUtil.entity.DTO);